phpBB/phpbb/mention/helper/mention_helper.php

```diff
@@ -0,0 +1,196 @@
+<?php
+/**
+ *
+ * phpBB mentions. A feature for the phpBB Forum Software package.
+ *
+ * @copyright (c) 2016, phpBB, https://www.phpbbextensions.io
+ * @license GNU General Public License, version 2 (GPL-2.0)
+ *
+ */
+
+namespace phpbb\mention\helper;
+
+use phpbb\db\driver\driver_interface;
+use phpbb\request\request_interface;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use phpbb\controller\helper;
+
+class mention_helper
+{
+    /**
+    * @var driver_interface
+    */
+    private $db;
+
+    /**
+    * @var Notification Details
+    */
+    private $data;
+
+    /**
+    * Mention Helper Class Constructor
+    *
+    * @param $db   Database Driver Interface Object.
+    *
+    * @return \phpbb\mention\helper\mention_helper
+    */
+    public function __construct(driver_interface $db)
+    {
+        $this->db = $db;
+    }
+
+    /**
+    * Function to extract the mentioned users using regex exp.
+    *
+    * @param string    $regular_expression_match     regex exp.
+    * @param string    $post_text                    Text pilled up from the board.
+    *
+    * @return array Consisiting of matches.
+    */
+    public function get_regex_match($regular_expression_match, $post_text)
+    {
+        $matches = false;
+        preg_match_all($regular_expression_match, $post_text, $matches, PREG_OFFSET_CAPTURE);
+        return $matches;
+    }
+
+    /**
+    * Function to extract the mentioned users using regex exp.
```

Comment:
same here
But what is the difference with the previous function? The comments are not clear